spacy/cli/debug_data.py

```diff
@@ -686,14 +757,23 @@ def _format_labels(
     return ", ".join([f"'{l}'" for l in cast(Iterable[str], labels)])
 
 
-def _get_examples_without_label(data: Sequence[Example], label: str) -> int:
+def _get_examples_without_label(
+    data: Sequence[Example], label: str, pipeline: str = "ner"
```

Comment:
Updated in https://github.com/explosion/spaCy/pull/10070/commits/9ed1d146d0688bb00b08b33d8bffb210cd231809

Reply:
Nevermind about this, see #10079 which will use `factory_name`.

Reply:
Ah, since `spancat` wasn't added yet I didn't look at this in #10079, but I think it would also need to use this as a factory name rather than a component name and have a separate implementation for `spancat` that includes the `spans_key`.

Reply:
Updated in https://github.com/explosion/spaCy/pull/10070/commits/3753a8b897a41d4942933c2885562df760188bcd
Not sure what's the recommended way to catch instances of `KeyError`... I was thinking of something like:

```python
try:
    ....
    eg.reference.spans[spans_key]
except KeyError:
   raise
```

But I don't see this pattern in the module itself.

Reply:
The reason you don't see this pattern anywhere is that we actively avoid it as part of our code style: https://github.com/explosion/spaCy/blob/master/extra/DEVELOPER_DOCS/Code%20Conventions.md#dont-use-tryexcept-for-control-flow

I'm not sure exactly what you're trying to do in this example, but I think you could just use `if spans_key in eg.reference.spans` instead.

I don't think anything in `debug data` should be actually raising a `KeyError` as part of the checks.

Reply:
Ah I see,  updated PR to include the check!